lab05-2.cpp

```diff
@@ -69,24 +69,24 @@ vector<string> vectorizeString(const string path)
 /****************************************
  * Create a canon token of a given path *
  ****************************************/
-vector<string> canonicalize(const std::string p)
+vector<string> canonicalize(const std::string p, const std::string base_path)
 {
-    string path = p, base_path = getBasePath();
+    string path = p;
     vector<string> vbase_path, vgiven_path;
     
     /*  Full path is given - starts with C/D drive
             GOOD: C:/User/Rodri/Test    */
-    if (path[0] == DRIVES[0] || path[0] == DRIVES[1])
+    if (path.length() > 1 && (path[0] == DRIVES[0] || path[0] == DRIVES[1]) && path[1] == ':')

```

Comment:
a length check to make sure we don't access invalid indexes
we need to check for ':' just in case there is a directory name that starts with 'C' or 'D'